### ISSUE TITLE
Test failure https://dev.azure.com/dnceng-public/public/_build/results?buildId=209919&view=ms.vss-test-web.build-test-results-tab&runId=3879622&resultId=162665&paneView=debug

### ISSUE THREAD

**Issue by v-wenyuxu** [dotnet/runtime#83659]
Run: [runtime-coreclr libraries-jitstress 20230319.1](https://dev.azure.com/dnceng-public/public/_build/results?buildId=209919&view=ms.vss-test-web.build-test-results-tab&runId=3879622&resultId=162665&paneView=debug)

Failed test:
```
net8.0-windows-Release-arm64-CoreCLR_checked-jitminopts-Windows.11.Arm64.Open

- System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: \"::$DATA\", alternateStream: \":bar:$DATA\")
- System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: \"::$DATA\", alternateStream: \":bar\")
- System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: \"\", alternateStream: \":bar:$DATA\")
- System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: \"\", alternateStream: \":bar\")
- System.IO.Tests.FileInfo_CopyTo_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: \"::$DATA\", alternateStream: \":bar:$DATA\")
... ...
```
**Error message:**
```
System.IO.IOException : The parameter is incorrect. : 'C:\h\w\A6E40946\t#File_Copy_str_str_b_bq13ehkl.mbj\WindowsAlternateDataStreamOverwrite_330_kaeldovu\WindowsAlternateDataStreamOverwrite_331_pdfk8999:bar:$DATA'


Stack trace
   at System.IO.FileSystem.CopyFile(String sourceFullPath, String destFullPath, Boolean overwrite) in /_/src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs:line 84
   at System.IO.Tests.File_Copy_str_str_b.Copy(String source, String dest, Boolean overwrite) in /_/src/libraries/System.IO.FileSystem/tests/File/Copy.cs:line 267
   at System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(String defaultStream, String alternateStream) in /_/src/libraries/System.IO.FileSystem/tests/File/Copy.cs:line 346
   at InvokeStub_File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(Object, Object, IntPtr*)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
```

**Comment by danmoseley** [comment]
Hi @v-wenyuxu thanks, please put test name not url in the title.

**Comment by Jozkee** [comment]
FWIW: There are other tests using alternate data stream paths and they didn't fail.
Only CopyFileEx with overwrite: true fails. I don't know what JitStress may be doing that affects this specific scenario.

**Comment by v-wenyuxu** [comment]
Failed again in: [runtime-coreclr libraries-jitstress 20230519.1](https://dev.azure.com/dnceng-public/public/_build/results?buildId=279301&view=ms.vss-test-web.build-test-results-tab&runId=5539570&resultId=174112&paneView=debug)

Failed test:
```
net8.0-windows-Release-arm64-CoreCLR_checked-zapdisable-Windows.11.Arm64.Open

- System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: \"::$DATA\", alternateStream: \":bar:$DATA\")
- System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: \"::$DATA\", alternateStream: \":bar\")
- System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: \"\", alternateStream: \":bar:$DATA\")
- System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: \"\", alternateStream: \":bar\")
```
**Error message:**
```
System.IO.IOException : The parameter is incorrect. : 'C:\h\w\BDDE0A0A\t#File_Copy_str_str_b_snfjrv1o.lza\WindowsAlternateDataStreamOverwrite_330_xpdes8ds\WindowsAlternateDataStreamOverwrite_331_2j4b3v36:bar:$DATA'



Stack trace
   at System.IO.FileSystem.CopyFile(String sourceFullPath, String destFullPath, Boolean overwrite) in /_/src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs:line 84
   at System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(String defaultStream, String alternateStream) in /_/src/libraries/System.IO.FileSystem/tests/File/Copy.cs:line 346
   at InvokeStub_File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(Object, Object, IntPtr*)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
```

**Comment by adamsitnik** [comment]
The error `The parameter is incorrect` usually means `ERROR_INVALID_PARAMETER` which means that and invalid value was passed to sys-call.

The logic around the sys-call seems very simple:

https://github.com/dotnet/runtime/blob/e91db04be24aac28fd041425fac014ef04d940b1/src/libraries/Common/src/Interop/Windows/Kernel32/Interop.CopyFile.cs#L11-L15

If we had a bug in System.IO, we would get it almost always. And it the provided logs I can see that it's throw on the second call to this sys-call (the first worked fine):

https://github.com/dotnet/runtime/blob/e91db04be24aac28fd041425fac014ef04d940b1/src/libraries/System.IO.FileSystem/tests/File/Copy.cs#L336-L346

Since it's specific to Windows arm64, I suspect that it's either a marshalling or runtime bug. @jkotas what do you think?

**Comment by AaronRobinsonMSFT** [comment]
> And it the provided logs I can see that it's throw on the second call to this sys-call (the first worked fine):
> Since it's specific to Windows arm64, I suspect that it's either a marshalling or runtime bug. 

@adamsitnik This is unlikely to an interop bug. I disagree with your analysis based on the above (2) statements.

Since all interop stub logic is idempotent, the first (1) and all subsequent operations run the same code. It is unlikely therefore that if the first succeeds and the second fails it has anything to do with the interop stub.

The second doesn't indicate anything to do with marshalling bug since interop generates IL and is platform agnostics - other than pointer size, in all cases I can think of.  Also, since this is passing on the first run the IL and all interop code is identical across runs.

It is entirely possible there is a codegen issue, which we should consider. However, this is File I/O and based on the in the failure it is entirely possible the Win32 File API is doing something the test is handling.

`'C:\h\w\BDDE0A0A\t#File_Copy_str_str_b_snfjrv1o.lza\WindowsAlternateDataStreamOverwrite_330_xpdes8ds\WindowsAlternateDataStreamOverwrite_331_2j4b3v36:bar:$DATA'`

In summary I don't think there is enough data to warrant sending this over to the interop team or any other runtime team at present. The owner of the File IO APIs should be able to run this on a Windows ARM64 machine and see if they can reproduce it and perhaps collect a bit more information.

**Comment by AaronRobinsonMSFT** [comment]
Ugh. This is a JIT stress issue. Nevermind, the codegen team needs to take a look.

**Comment by AaronRobinsonMSFT** [comment]
@dotnet/jit-contrib Is there anything the owner of this test could collect that would help with the investigation?

**Comment by jakobbotsch** [comment]
As I pointed out above the failure was in the zapdisable stress mode and the recent one is in jitminopts. There's no abnormal jit stress enabled in these scenarios; the first one is like running without R2R turned on, the second one is like running debug code gen. Is there any reason to believe this is a codegen issue? Have we tried reproducing the issue with the paths involved?

**Comment by jkotas** [comment]
> @jkotas what do you think?

Based on the data presented so far, it is most likely Windows Arm64 specific OS issues. area-System.IO owners should investigate it and file a bug against the OS team as appropriate.

Also, I would check for potential machine setup issue. It is possible that the file system is misconfigured on some of the Windows Helix machines that makes the test fail.

**Comment by adamsitnik** [comment]
Thank you all for your feedback, I am going to try to repro it on my Windows x64 and arm64 machines.

**Comment by BruceForstall** [comment]
@Jozkee @adamsitnik This issue is marked as both "blocking-clean-ci-optional" and "Milestone: Future". That doesn't make sense. If it's actually blocking, it should be in the current release milestone. If it's not blocking, remove the blocking label.

**Comment by agocke** [comment]
This is failing in Native AOT now as well. Since jitstress saw this first it would be good if someone from JIT could take a look.

**Comment by jakobbotsch** [comment]
@agocke Do you have any new evidence to suggest this is a JIT issue? I do not believe this is a JIT issue -- please see the comments above for context.

**Comment by jeffhandley** [comment]
Added this to the 9.0.0 milestone because of the frequency

**Comment by carlossanlop** [comment]
I adjusted the KnownBuildError pattern. It was not catching many hits.

I found this in a 7.0 PR: https://github.com/dotnet/runtime/pull/98223 

- Queue: `Build windows arm64 Release NativeAOT`
- Job result: https://dev.azure.com/dnceng-public/public/_build/results?buildId=560133&view=logs&j=7bb91cf6-d57c-5393-cf5c-c884f1e50962&t=7729b4d4-9255-54d8-01fd-e831ebbb744b&l=57
- Log file: https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-98223-merge-f061765a12904563bb/System.IO.FileSystem.Tests/1/console.c445635c.log?helixlogtype=result
- Output:
```
[FAIL] System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: "", alternateStream: ":bar")
System.IO.IOException : The parameter is incorrect. : 'C:\h\w\9BDA087A\t\#File_Copy_str_str_b_uhztkjkv.b14\WindowsAlternateDataStreamOverwrite_330_3ah48qd5\WindowsAlternateDataStreamOverwrite_331_4vafo98k:bar'
   at System.IO.FileSystem.CopyFile(String, String, Boolean) + 0xf8
   at System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(String defaultStream, String alternateStream) + 0x240
   at System.IO.FileSystem.Tests!<BaseAddress>+0xa03d4c
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0x150
[FAIL] System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: "", alternateStream: ":bar:$DATA")
System.IO.IOException : The parameter is incorrect. : 'C:\h\w\9BDA087A\t\#File_Copy_str_str_b_w5u25uum.v1y\WindowsAlternateDataStreamOverwrite_330_1eewcxsj\WindowsAlternateDataStreamOverwrite_331_uyotey4z:bar:$DATA'
   at System.IO.FileSystem.CopyFile(String, String, Boolean) + 0xf8
   at System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(String defaultStream, String alternateStream) + 0x240
   at System.IO.FileSystem.Tests!<BaseAddress>+0xa03d4c
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0x150
[FAIL] System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: "::$DATA", alternateStream: ":bar")
System.IO.IOException : The parameter is incorrect. : 'C:\h\w\9BDA087A\t\#File_Copy_str_str_b_dwo5dugm.mu0\WindowsAlternateDataStreamOverwrite_330_0zin5tx0\WindowsAlternateDataStreamOverwrite_331_589zzmte:bar'
   at System.IO.FileSystem.CopyFile(String, String, Boolean) + 0xf8
   at System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(String defaultStream, String alternateStream) + 0x240
   at System.IO.FileSystem.Tests!<BaseAddress>+0xa03d4c
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0x150
[FAIL] System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: "::$DATA", alternateStream: ":bar:$DATA")
System.IO.IOException : The parameter is incorrect. : 'C:\h\w\9BDA087A\t\#File_Copy_str_str_b_z0jr4yj4.rn3\WindowsAlternateDataStreamOverwrite_330_seuj2fuy\WindowsAlternateDataStreamOverwrite_331_v1a2zn1a:bar:$DATA'
   at System.IO.FileSystem.CopyFile(String, String, Boolean) + 0xf8
   at System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(String defaultStream, String alternateStream) + 0x240
   at System.IO.FileSystem.Tests!<BaseAddress>+0xa03d4c
   at System.Reflection.DynamicInvokeInfo.Invoke(Object, IntPtr, Object[], BinderBundle, Boolean) + 0x150
```

**Comment by jkotas** [comment]
Hit in https://github.com/dotnet/runtime/pull/100233

**Comment by carlossanlop** [comment]
@dotnet/area-system-io  An unrealted deps flow PR hit this failure in 8.0:

- PR: https://github.com/dotnet/runtime/pull/101941
- Affected queue: `windows-arm64 Release NativeAOT_Libraries`
- Build result: https://dev.azure.com/dnceng-public/public/_build/results?buildId=675511&view=logs&j=b0477d29-a3e8-5063-350d-e2bc4b70853a&t=cc495a94-4eb4-54d1-d51d-36ad20b1ec33&l=56
- Log file: https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-101941-merge-4358b6180a90427aa4/System.IO.FileSystem.Tests/1/console.014d33bb.log?helixlogtype=result
- Output:
```
[FAIL] System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: "", alternateStream: ":bar")
System.IO.IOException : The parameter is incorrect. : 'C:\h\w\A0100912\t\#File_Copy_str_str_b_qheie0wz.qb1\WindowsAlternateDataStreamOverwrite_348_ccrsp4uq\WindowsAlternateDataStreamOverwrite_349_8tvqm5y1:bar'
   at System.IO.FileSystem.CopyFile(String, String, Boolean) + 0xf4
   at System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(String defaultStream, String alternateStream) + 0x260
   at System.IO.FileSystem.Tests!<BaseAddress>+0x9e7694
   at System.Reflection.DynamicInvokeInfo.InvokeWithFewArguments(IntPtr, Byte&, Byte&, Object[], BinderBundle, Boolean) + 0x78
[FAIL] System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: "", alternateStream: ":bar:$DATA")
System.IO.IOException : The parameter is incorrect. : 'C:\h\w\A0100912\t\#File_Copy_str_str_b_402xd4nr.tat\WindowsAlternateDataStreamOverwrite_348_4ybuuzqq\WindowsAlternateDataStreamOverwrite_349_1b12cs7h:bar:$DATA'
   at System.IO.FileSystem.CopyFile(String, String, Boolean) + 0xf4
   at System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(String defaultStream, String alternateStream) + 0x260
   at System.IO.FileSystem.Tests!<BaseAddress>+0x9e7694
   at System.Reflection.DynamicInvokeInfo.InvokeWithFewArguments(IntPtr, Byte&, Byte&, Object[], BinderBundle, Boolean) + 0x78
[FAIL] System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: "::$DATA", alternateStream: ":bar")
System.IO.IOException : The parameter is incorrect. : 'C:\h\w\A0100912\t\#File_Copy_str_str_b_r42qnfwb.cf5\WindowsAlternateDataStreamOverwrite_348_lam2jk4y\WindowsAlternateDataStreamOverwrite_349_v4camptt:bar'
   at System.IO.FileSystem.CopyFile(String, String, Boolean) + 0xf4
   at System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(String defaultStream, String alternateStream) + 0x260
   at System.IO.FileSystem.Tests!<BaseAddress>+0x9e7694
   at System.Reflection.DynamicInvokeInfo.InvokeWithFewArguments(IntPtr, Byte&, Byte&, Object[], BinderBundle, Boolean) + 0x78
[FAIL] System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(defaultStream: "::$DATA", alternateStream: ":bar:$DATA")
System.IO.IOException : The parameter is incorrect. : 'C:\h\w\A0100912\t\#File_Copy_str_str_b_tfe1qroe.1lr\WindowsAlternateDataStreamOverwrite_348_iiyqfmza\WindowsAlternateDataStreamOverwrite_349_6927o4x5:bar:$DATA'
   at System.IO.FileSystem.CopyFile(String, String, Boolean) + 0xf4
   at System.IO.Tests.File_Copy_str_str_b.WindowsAlternateDataStreamOverwrite(String defaultStream, String alternateStream) + 0x260
   at System.IO.FileSystem.Tests!<BaseAddress>+0x9e7694
   at System.Reflection.DynamicInvokeInfo.InvokeWithFewArguments(IntPtr, Byte&, Byte&, Object[], BinderBundle, Boolean) + 0x78
```

**Comment by kunalspathak** [comment]
Hit in #107766

**Comment by jeffschwMSFT** [comment]
removing blocking-clean-ci as this has not failed in 30 days

24-Hour Hit Count | 7-Day Hit Count | 1-Month Count
-- | -- | --
0 | 0 | 0
